nncf/experimental/openvino_native/graph/model_transformer.py

```diff
@@ -87,41 +90,49 @@ def transform(self, transformation_layout: TransformationLayout) -> ov.Model:
                 model_extraction_transformation = transformation
             elif isinstance(transformation, OVBiasCorrectionCommand):
                 bias_correction_transformations.append(transformation)
-
-        if output_insertion_transformations:
-            self._apply_output_insertion_transformations(output_insertion_transformations)
+        model = None
+        # Inplace transformations; Using deepcopy of model
         if fq_nodes_removing_transformations:
-            self._apply_fq_nodes_removing_transformation(fq_nodes_removing_transformations)
+            model = self._apply_fq_nodes_removing_transformation(fq_nodes_removing_transformations)
         if quantizer_insertion_transformations:
-            self._apply_quantizer_insertion_transformations(quantizer_insertion_transformations)
+            model = self._apply_quantizer_insertion_transformations(quantizer_insertion_transformations)
         if bias_correction_transformations:
-            self._apply_bias_correction_transformations(bias_correction_transformations)
+            model = self._apply_bias_correction_transformations(bias_correction_transformations)
         if model_extraction_transformation:
-            self._model = self._apply_model_extraction_transformation(model_extraction_transformation)
-
-        return self._model
-
-    def _apply_output_insertion_transformations(self, transformations: List[OVOutputInsertionCommand]) -> None:
+            model = self._apply_model_extraction_transformation(model_extraction_transformation)
+        if output_insertion_transformations:
+            model = self._apply_output_insertion_transformations(output_insertion_transformations)
+        # No transformation applied
+        if model is None:
+            logger.warning('No transformations were applied to the model. The copy of the model is returned.')
+            return self._model.clone()
```

Comment:
Why we should return a copy?

Reply:
The function should have only one expected behavior to not ruin the user pipeline.
The user expects to get a new 'transformed' model from the transform function to store two models in the pipeline, e.g. performance measurement and etc. he 'transformed' model can be changed in the user pipeline, and it should not affect the original model.

Reply:
What does
```python
self.onnx_model_extractor.extract_model(transformation.inputs, transformation.outputs)
```
return? If I change the extracted submodel the original model will be changed?

Reply:
It creates a new model and doesn't change the original